scheds/rust/scx_rustland/src/main.rs

```diff
@@ -407,18 +389,27 @@ impl<'a> Scheduler<'a> {
     fn dispatch_tasks(&mut self) {
         let maps = self.skel.maps();
         let dispatched = maps.dispatched();
+        let idle_cpus = self.get_idle_cpus();
 
         // Dispatch only a batch of tasks equal to the amount of idle CPUs in the system.
         //
         // This allows to have more tasks sitting in the task pool, reducing the pressure on the
         // dispatcher queues and giving a chance to higher priority tasks to come in and get
         // dispatched earlier, mitigating potential priority inversion issues.
-        for _ in 0..self.get_idle_cpus() {
+        for cpu in &idle_cpus {
             match self.task_pool.pop() {
-                Some(task) => {
+                Some(mut task) => {
                     // Update global minimum vruntime.
                     self.min_vruntime = task.vruntime;
 
+                    // Select a CPU to dispatch the task.
+                    //
+                    // Use the previously used CPU if idle, that is always the best choice (to
+                    // mitigate migration overhead), otherwise pick the next idle CPU available.
+                    if !idle_cpus.contains(&task.cpu) {
+                        task.cpu =  *cpu;
+                    }
```

Comment:
This still can lead to multiple tasks being assigned to the same cpu, right? Something to improve in the future, I guess.

Reply:
@htejun exactly, we can still assign the same idle cpu to multiple tasks if their `prev_cpu` is still idle, that is not super bad, being too aggressive with migrations also comes with a cost. We could also speculate on the fact that tasks with a close vruntime should be spread across multiple CPUs, so that they can run pretty much at the same time in parallel, rather than piling them up on the same idle CPU (that is also something that I'd like to experiment).

But there is definitely room for improvements in the cpu selection logic and in the userspace scheduler in general.

And I'm happy to see that we are reaching a stage where the BPF component is becoming a more "stable" backend (requiring minimal changes), so that we can start channeling all the optimizations, improvements, tests and experiments in the user-space part. If that's the case it would be a nice success story for this little project IMHO. :)